src/main/java/catan/services/util/play/PlayUtil.java

```diff
@@ -0,0 +1,135 @@
+        }
+
+        log.debug("Next move order in {} stage is changing from {} to {}", game.getStage(), game.getCurrentMove(), nextMoveNumber);
+        game.setCurrentMove(nextMoveNumber);
+    }
+
+    public void updateCurrentCycleBuildingNumber(GameBean game) {
+        if (game.getStage().equals(GameStage.PREPARATION)) {
+            List<List<String>> initialBuildingsSet = getInitialBuildingsSetFromJson(game.getInitialBuildingsSet());
+            if (game.getCurrentCycleBuildingNumber() > initialBuildingsSet.get(game.getPreparationCycle() - 1).size()) {
+                game.setCurrentCycleBuildingNumber(1);
+            } else {
+                game.setCurrentCycleBuildingNumber(game.getCurrentCycleBuildingNumber() + 1);
+            }
+        }
+    }
+    public void updateAvailableUserActions(GameBean game) throws GameException {
+        boolean isMandatory = false;
+        List<String> actionsList = new ArrayList<String>();
+        List<GameUserActions> actionCode = new ArrayList<GameUserActions>();
+        switch (game.getStage()) {
+            case PREPARATION:
+                List<List<String>> initialBuildingsSet = getInitialBuildingsSetFromJson(game.getInitialBuildingsSet());
+                for (GameUserBean gameUser : game.getGameUsers()) {
+                    if (gameUser.getMoveOrder() == game.getCurrentMove()) {
+                        actionCode = getActionCodeInPreparationStage(game, initialBuildingsSet);
+                    }
+                    if (!actionCode.isEmpty()) {
+                        actionsList.add("{\"code\": \"" + actionCode.get(0).toString() + "\", \"params\": {}}");
+                        gameUser.setAvailableActions("{\"list\": [" + actionsList.get(0) + "], \"isMandatory\": true}");
+                    } else {
+                        gameUser.setAvailableActions("{\"list\": [], \"isMandatory\": false}");
+                    }
+                    actionCode.clear();
+                    actionsList.clear();
+                }
+                break;
+            case MAIN:
+                //TODO: complete this part when developing main stage part
+                for (GameUserBean gameUser : game.getGameUsers()) {
+                    if (gameUser.getMoveOrder() == game.getCurrentMove()) {
+                        gameUser.setAvailableActions("{\"list\": [], \"isMandatory\": false}");
+                    } else {
+                        gameUser.setAvailableActions("{\"list\": [], \"isMandatory\": false}");
+                        //non-active users have not any available actions (trade in future)
+                    }
+                }
+                break;
+            default:
+                log.debug("Cannot recognize current game stage: {}", game.getStage());
+                throw new GameException(ERROR_CODE_ERROR);
+        }
+    }
+
+    private List<GameUserActions> getActionCodeInPreparationStage (GameBean game, List<List<String>> initialBuildingsSet) {
```

Comment:
enum class renamed to GameUserActionCode
and method to getListOfActionCodesInPreparationStage